astropy/io/fits/header.py

```diff
@@ -1914,13 +1903,8 @@ def _wildcardmatch(self, pattern):
         """
         pattern = pattern.replace("*", r".*").replace("?", r".")
         pattern = pattern.replace("...", r"\S*") + "$"
-        pattern_re = re.compile(pattern, re.I)
-
-        return [
-            idx
-            for idx, card in enumerate(self._cards)
-            if pattern_re.match(card.keyword)
-        ]
+        pattern = re.compile(pattern, re.I)
```

Comment:
To not overwrite pattern, maybe `match_pattern = re.compile(pattern, re.I).match`?